pkg/reconciler/utils/configmap.go

```diff
@@ -57,8 +57,12 @@ func (r *ConfigMapReconciler) ReconcileConfigMap(obj runtime.Object, cm *corev1.
 	if err != nil {
 		return nil, err
 	}
-
-	if !equality.Semantic.DeepEqual(cm.BinaryData, current.BinaryData) || !equality.Semantic.DeepEqual(cm.Data, current.Data) {
+	if eqFunc == nil {
+		eqFunc = func(cm1, cm2 *corev1.ConfigMap) bool {
+			return equality.Semantic.DeepEqual(cm1.BinaryData, cm2.BinaryData) && equality.Semantic.DeepEqual(cm1.Data, cm2.Data)
```

Comment:
The default eqFunc should only look at cm.Data only

Reply:
In this case, eqFunc only looking at cm.Data would also fix #1611 . Do we still want to test for a difference in cm.BinaryData for TargetsConfig, even if looking only at cm.Data suffices to fix the bug?

Reply:
cm.Data is meant for debugging purpose and we may get rid of it in the future to reduce configmap size, see #964.

Reply:
Ack. Fixed.